test/entity/entity_test.go

```diff
@@ -0,0 +1,222 @@
+			entityName:              func(v string) bool { return v == ev.expectedEntity.name },
+			entityEnvironment:       func(v string) bool { return v == ev.expectedEntity.environment },
+			entityPlatform:          func(v string) bool { return v == ev.expectedEntity.platformType },
+			entityEKSCluster:        func(v string) bool { return v == ev.expectedEntity.eksCluster },
+			entityK8sNode:           func(v string) bool { return v == ev.expectedEntity.k8sNode },
+			entityK8sNamespace:      func(v string) bool { return v == ev.expectedEntity.k8sNamespace },
+			entityK8sWorkload:       func(v string) bool { return v == ev.expectedEntity.k8sWorkload },
+			entityServiceNameSource: func(v string) bool { return v == ev.expectedEntity.serviceNameSource },
+		}
+	}
+	return ev
+}
+
+// ValidateField checks if a field is expected and matches the expected value.
+func (ev *EntityValidator) ValidateField(field, value string, t *testing.T) {
+	if validator, ok := ev.fieldValidators[field]; ok {
+		ev.requiredFields[field] = true
+		assert.True(t, validator(value), "Validation failed for field %s", field)
+	}
+}
+
+// AllFieldsPresent ensures all required fields are found.
+func (ev *EntityValidator) AllFieldsPresent() bool {
+	for _, present := range ev.requiredFields {
+		if !present {
+			return false
+		}
+	}
+	return true
+}
+
+type expectedEntity struct {
+	entityType        string
+	name              string
+	environment       string
+	platformType      string
+	k8sWorkload       string
+	k8sNode           string
+	k8sNamespace      string
+	eksCluster        string
+	serviceNameSource string
+	instanceId        string
+}
+
+func init() {
+	environment.RegisterEnvironmentMetaDataFlags()
+}
+
+// TestPutLogEventEntityEKS checks if entity is emitted correctly in EKS
+// through FluentBit
+func TestPutLogEventEntityEKS(t *testing.T) {
+	var instancePrivateDNS *string
+
+	env := environment.GetEnvironmentMetaData()
+	assert.NotEmpty(t, env.InstanceId)
+	assert.NotEmpty(t, env.EKSClusterName)
+	assert.NotEmpty(t, env.ComputeType)
+	if env.InstanceId != "" {
+		var err error
+		instancePrivateDNS, err = awsservice.GetInstancePrivateIpDns(env.InstanceId)
+		assert.NoError(t, err)
+		assert.NotEmpty(t, instancePrivateDNS)
+	}
+
+	// ensure that there is enough time from the "start" time and the first log line,
+	// so we don't miss it in the StartQuery call
+	log.Printf("Sleeping for %f seconds to ensure log group is ready for query", sleepForFlush.Seconds())
+	time.Sleep(sleepForFlush)
+	end := time.Now()
+
+	testCases := map[string]struct {
+		agentConfigPath string
+		podName         string
+		useEC2Tag       bool
+		expectedEntity  expectedEntity
+	}{
+		"Entity/K8sWorkloadServiceNameSource": {
+			agentConfigPath: filepath.Join("resources", "compass_default_log.json"),
+			podName:         "log-generator",
+			expectedEntity: expectedEntity{
+				entityType:        "Service",
```

Comment:
nit: we could have an enum for different entity types: Service, Resource

Same for platformType: AWS::EKS, AWS::K8s, etc